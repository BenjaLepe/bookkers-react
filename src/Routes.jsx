import React from 'react';
import { Route, Switch } from 'react-router-dom';
import NotFound from './views/NotFound';
import Login from './views/Login';
import BookDetail from './views/BookDetail';
import SignUp from './views/SignUp';
import Home from './views/Home';
import AuthContextProvider from './contexts/AuthContext';
import SessionProfile from './views/SessionProfile';
import SessionExpired from './views/SessionExpired';
import UserProfile from './views/UserProfile';
import CreateBook from './views/AddBook';
import EditUser from './views/EditUser';
import Navbar from './Components/Navbar';
import reviewReports from './views/Reports';
import EditBook from './views/EditBook';

export default function Routes() {
  return (
    <AuthContextProvider>
      <Navbar />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/login" component={Login} />
        <Route exact path="/sign-up" component={SignUp} />
        <Route exact path="/session-expired" component={SessionExpired} />
        <Route exact path="/session-profile" component={SessionProfile} />
        <Route exact path="/users/:id" component={UserProfile} />
        <Route exact path="/books/new" component={CreateBook} />
        <Route exact path="/books/:id" component={BookDetail} />
        <Route exact path="/edit-book/:id" component={EditBook} />
        <Route exact path="/edit-user/:id" component={EditUser} />
        <Route exact path="/books/:book_id/reviews/:review_id/reports" component={reviewReports} />
        <Route component={NotFound} />
      </Switch>
    </AuthContextProvider>
  );
}