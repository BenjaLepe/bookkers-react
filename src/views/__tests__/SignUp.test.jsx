import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Routes from '../../Routes';

const { create } = renderer;

function TestRouter({ path }) {

  return (
    <MemoryRouter initialEntries={[path]}>
      <Routes />
    </MemoryRouter>
  );
}

describe('SignUp', () => {
  describe('When there is no current session', () => {
    it ('renders the sign up form.', () => {
      const DOM = create(<TestRouter path ="/sign-up" />).toJSON();
      expect(DOM).toMatchSnapshot();
    });
  });

  describe('When there is a current session', () => {
    const user = {
      access_token: 'TestToken',
      token_type: 'Bearer',
      user: {
        username: 'testUserName',
        first_name: 'testFirstName',
        last_name: 'testLastName',
        email: 'test@email.cl',
        description: 'testDescription',
      },
    };
    const sessionExp = new Date(
      new Date().getTime() + 1000 * 60 * 60 * 24, // One day in the future
    );
    const localStorageData = {
      user,
      sessionExp,
    }

    beforeEach(() => {
      global.Storage.prototype.getItem = jest.fn(
        (key) => JSON.stringify(localStorageData[key]),
      );
    });

    afterEach(() => {
      global.Storage.prototype.getItem.mockReset();
    });

    it('renders the home page with books', () => {
      const DOM = create(<TestRouter path="/sign-up" />).toJSON();
      expect(DOM).toMatchSnapshot();
    });
  });
});