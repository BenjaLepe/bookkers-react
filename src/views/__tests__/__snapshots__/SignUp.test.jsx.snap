// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp When there is a current session renders the home page with books 1`] = `
Array [
  <div
    className="nav-bg"
  >
    <div
      className="nav-bar container"
    >
      <div
        className="nav-group-logo"
      >
        <a
          aria-current="page"
          className="nav-link nav-title active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Bookkers
        </a>
      </div>
      <div
        className="nav-group"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/session-profile"
          onClick={[Function]}
        >
          Profile
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/books/new"
          onClick={[Function]}
        >
          Add book
        </a>
        <button
          className="nav-button"
          onClick={[Function]}
          type="button"
        >
          Log out
        </button>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="head-container"
    >
      <div
        className="logo-container"
      >
        <img
          alt="LOGO"
          className="logo"
          src="logo.png"
        />
      </div>
      <div
        className="welcome-container"
      >
        <h2>
          Welcome 
          testFirstName
           to Bookers
        </h2>
        <p>
          Bookkers es una red social que permite a sus usuarios publicar, calificar, recomendar y ver las recomendaciones que otros usuarios han hecho respecto a los libros ingresados a la aplicaci√≥n.
        </p>
      </div>
    </div>
    <div
      className="container"
    >
      <h2>
        Books
      </h2>
      <div
        className="books-collection"
      />
      <div
        className="page-set"
      >
        <button
          className="page-button"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Back
        </button>
        <p
          className="pages"
        >
          1 : 1
        </p>
        <button
          className="page-button"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`SignUp When there is no current session renders the sign up form. 1`] = `
Array [
  <div
    className="nav-bg"
  >
    <div
      className="nav-bar container"
    >
      <div
        className="nav-group-logo"
      >
        <a
          aria-current="page"
          className="nav-link nav-title active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Bookkers
        </a>
      </div>
      <div
        className="nav-group"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Log In
        </a>
        <a
          aria-current="page"
          className="nav-link active"
          href="/sign-up"
          onClick={[Function]}
          style={Object {}}
        >
          Sign Up 
        </a>
      </div>
    </div>
  </div>,
  <div>
    <form
      action="#"
      className="form-box"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <h2>
        Create a new account
      </h2>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="username"
          >
            Username:
          </label>
          <br />
        </div>
        <div
          className="field-container"
        >
          <input
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="first_name"
          >
            First name:
          </label>
          <br />
        </div>
        <div
          className="field-container"
        >
          <input
            name="first_name"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="last_name"
          >
            Last name:
          </label>
          <br />
        </div>
        <div
          className="field-container"
        >
          <input
            name="last_name"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="email"
          >
            Email:
          </label>
          <br />
        </div>
        <div
          className="field-container"
        >
          <input
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="password"
          >
            Password:
          </label>
          <br />
        </div>
        <div
          className="field-container"
        >
          <input
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="passwordConfirmation"
          >
            Confirm password:
          </label>
          <br />
        </div>
        <div
          className="field-container"
        >
          <input
            name="passwordConfirmation"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="profile_picture"
          >
            Profile picture:
          </label>
          <br />
        </div>
        <div
          className="field-container"
        >
          <input
            name="profile_picture"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            as="textarea"
            htmlFor="description"
          >
            Description:
          </label>
          <br />
        </div>
        <div
          className="field-container"
        >
          <input
            name="description"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="submit-container"
      >
        <button
          type="submit"
        >
          Sign Up
        </button>
      </div>
    </form>
    <p>
      
    </p>
  </div>,
]
`;
